src/hooks/usePagination.js

```diff
@@ -1,14 +1,35 @@
 export const DOTS = "...";
 
-function usePagination() {
-  /*
-    Rewrite the logic here to map out the pagination to be displayed
-
-    !!!!!! ATTENTION !!!!!!
-    Please replace this comment here with a description of this hook.
-    
-  */
-  return [1, 2, 3, DOTS, 5];
+function usePagination(object) {
+  const { currentPage, totalCount, pageSize } = object;
+  const pageIncrement = 1;
+  const rightSibling = currentPage + pageIncrement;
+  const leftSibling = currentPage - pageIncrement;
+  const maxPage = Math.ceil(totalCount / pageSize);
+  const pageRange = [];
+  const pageWithDots = [];
+  let previousNumber;
+
+  for (let i = 1; i <= maxPage; i++) {
```

Comment:
Looping over all the possible page numbers is inefficient. Can you refactor the code to avoid this loop.